007-design-docs.md

```diff
@@ -0,0 +1,86 @@
+
+#### Strimzi Design Document Organization
+Design documents should be split into logical areas, for repositories that aren't `strimzi-kafka-operator` I suspect this would start as a single or just a few succinct documents.
+For `strimzi-kafka-operator` I would suggest splitting design documentation from a functionality perspective, that is to say all documents relating to the `Kafka` reconcile and custom resource would be separated into its own folder called `kafka.md` or `kafka-operator.md`, similarly for `KafkaConnect` etc.
+For design documentation that would cover the whole operator and not just any single custom resource I would suggest a file named `operator.md` this would describe behaviours not unique to any one specific reconcile loop or verticle and appropriate links to other files containing specific (but still high-level) differences between reconciles could be described there. 
+
+
+### How to write a design document
+
+The design documents would live in the related repository in the Strimzi Organization and would match the current code state. Naturally it is hard to align the design documents and the code behaviour, but it would be the responsibility of a reviewer to identify if any of the changes are impactful enough to potentially affect the design docs. Due to the below described simplicity of the design docs, if they are found to be out of date, it should be fairly simply to raise a pull request with the behavioural changes documented.
+
+A design document would be split into the following parts:
+
+**Title**
+
+The title of the design document, normally this would pertain to a single component or one larger aspect of the whole of Strimzi. New pages inclusion (but not contents) would be discussed and decided on as part of a proposal if applicable, or during development if the complexity of a component merits a design document as judged by the maintainers.
+
+**Overview, Context and Description**
+
+A high level summary that any user of Strimzi should be able understand and use to decide if it’s useful for them to read the rest of the doc.
+A description of the problem the component/code tackles, why the program/code is necessary, what people need to know to assess the success of the implementation.
+
+**Configuration**
+Describe how users interact with this component or system. I.e. configured via envars, custom resources, REST calls...
+This information will be invaluable to a reader to understand where the configuration points are and how to interact with the system/component.
+
+**Dependencies and Scope**
+Is this intended to be a standalone component, are other components dependent on it, or is it dependent on other components.
+Is this intended for a single Kafka cluster or to interact with multiple.
+
+
+### When is a design document complete
+
+A design document is considered complete when the reviewers and maintainers believe the document to accurately reflect the implementation of the components behaviour. 
```

Comment:
I would probably remove this sentence. I think that at the end the document is never complete because the code evolves and the document has to evolve together with it.